internal/xpkg/v2/parser/linter/linter.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package linter contains utilities for linting packages.
+package linter
+
+import (
+	"strings"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+)
+
+const (
+	errNilLinterFn = "linter function is nil"
+
+	errOrFmt = "object did not pass any of the linters with following errors: %s"
+)
+
+// TODO(@tnthornton) update API in crossplane-runtime to use Package interface
+// instead of concreate type and remove this package.
```

Comment:
Something that might be easier to clean up in a future PR 👍.

FWIW, I believe most of this file is a copy of https://github.com/crossplane/crossplane-runtime/blob/c05e74a4b5cf7ce0c42dedf46f5126e71692958f/pkg/parser/linter.go with the exception of the Package declaration below.

Reply:
I [added](https://github.com/crossplane/crossplane-runtime/pull/553) it already to crossplane-runtime, but slightly different name not to break the crossplane-runtime api. Will deduplicate it later on!